playwright/_impl/_browser.py

```diff
@@ -106,7 +108,9 @@ async def new_context(
         storageState: Union[StorageState, str, Path] = None,
     ) -> BrowserContext:
         params = locals_to_params(locals())
-        normalize_context_params(self._connection._is_sync, params)
+        await run_in_threadpool(
```

Comment:
The author of normalize_context_params might be surprised with the fact it is called on another thread. You should make `normalize_context_params` async instead and use the executor call from inside.